Content.Shared/_CM14/Scoping/SharedScopeSystem.User.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Robust.Shared.Containers;
+using Robust.Shared.Player;
+
+namespace Content.Shared._CM14.Scoping;
+
+public abstract partial class SharedScopeSystem
+{
+    public void InitializeUser()
+    {
+        SubscribeLocalEvent<ScopeUserComponent, MoveEvent>(OnMove);
+        SubscribeLocalEvent<ScopeUserComponent, EntParentChangedMessage>(OnParentChanged);
+        SubscribeLocalEvent<ScopeUserComponent, ContainerGettingInsertedAttemptEvent>(OnInsertAttempt);
+        SubscribeLocalEvent<ScopeUserComponent, EntityTerminatingEvent>(OnEntityTerminating);
+
+        SubscribeLocalEvent<ScopeUserComponent, PlayerDetachedEvent>(OnPlayerDetached);
+    }
+
+    private void OnMove(EntityUid uid, ScopeUserComponent component, ref MoveEvent args)
+    {
+        if (!TryComp<ScopeComponent>(component.ScopingItem, out var scopeComponent))
+            return;
+
+        if (!_transformSystem.InRange(scopeComponent.LastScopedAt, args.NewPosition, 1.5f))
+            UserStopScoping(uid, component);
+    }
+
+    private void OnParentChanged(EntityUid uid, ScopeUserComponent component, ref EntParentChangedMessage args)
+    {
+        UserStopScoping(uid, component);
+    }
+
+    private void OnInsertAttempt(EntityUid uid, ScopeUserComponent component, ContainerGettingInsertedAttemptEvent args)
+    {
+        UserStopScoping(uid, component);
+    }
```

Comment:
Are these two intended for cases when the rifle is dropped from the user's hands? If so `UnequippedHandEvent` should suffice.
```suggestion
    private void OnParentChanged(EntityUid uid, ScopeUserComponent component, ref EntParentChangedMessage args)
    {
        UserStopScoping(uid, component);
    }

    private void OnInsertAttempt(EntityUid uid, ScopeUserComponent component, ContainerGettingInsertedAttemptEvent args)
    {
        UserStopScoping(uid, component);
    }
```

Reply:
Those are intended to be triggered when the user is put in a container (felinid) or a box, would only `OnParentChanged` suffice?

Reply:
Ah I see the code to which this comment was meant, will fix